### PR TITLE
Refactored core and SoapUIClient to support RecipeExecutionResult

### PR DESCRIPTION
**Pull request by olensmar**
I've refactored a bunch to support the RecipeExecutionResult related functionality in the SoapUI client - much of this is extracting common functionality to interfaces and shared abstract base classes - and then implementing these for both TestServer and SoapUI. This will probably not be backwards compatible so I suggest we bump the version to either 1.3.0-SNAPSHOT or 2.0.0-SNAPSHOT

### CODE REVIEW

File: soapui-local-client/src/main/java/com/smartbear/readyapi/client/execution/SoapUIRecipeExecutor.java [hunk 1]
```diff
@@ -57,23 +92,92 @@
...
+    }
+
+    @Override
+    public Execution executeRecipe(TestRecipe recipe) throws ApiException {
+        return postTestCase(recipe.getTestCase(), false);
+    }
+
+    @Override
+    public List<Execution> getExecutions() throws ApiException {
+        return Lists.newArrayList(executionsMap.values());
```
Comment (spindelmanne) [inline soapui-local-client/src/main/java/com/smartbear/readyapi/client/execution/SoapUIRecipeExecutor.java]: Use new ArrayList<>(...) instead.

File: ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/execution/TestServerRecipeExecutionResult.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.smartbear.readyapi.client.execution;
+
+import com.smartbear.readyapi.client.model.ProjectResultReport;
+import com.smartbear.readyapi.client.result.AbstractRecipeExecutionResult;
+import com.smartbear.readyapi.client.result.TestServerTestStepResult;
+
+public class TestServerRecipeExecutionResult extends AbstractRecipeExecutionResult {
```
Comment (spindelmanne) [inline ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/execution/TestServerRecipeExecutionResult.java]: We should really have at least class-level Javadoc everywhere. Remember this is supposed to be a public API; that makes it even more important!

File: ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/execution/Execution.java [hunk 3]
```diff
@@ -1,201 +1,20 @@
...
-import io.swagger.client.auth.HttpBasicAuth;
 
-import java.util.Collections;
-import java.util.Deque;
 import java.util.List;
-import java.util.concurrent.ConcurrentLinkedDeque;
 
-/**
- * Represents an execution, synchronous or asynchronous, and encapsulates all available information about the execution.
- */
```
Comment (olensmar) [inline ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/execution/Execution.java]: by mistake...

File: ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/result/AbstractRecipeExecutionResult.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+    @Override
+    public TestStepResult getFirstTestStepResult(String name) {
+        for (TestStepResult testStepResultReport : results) {
+            if (testStepResultReport.getTestStepName().equalsIgnoreCase(name)) {
+                return testStepResultReport;
+            }
+        }
+
+        return null;
```
Comment (olensmar) [inline ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/result/AbstractRecipeExecutionResult.java]: yes

File: soapui-local-client/src/main/java/com/smartbear/readyapi/client/execution/SoapUIRecipeExecutor.java (hunk 1 above)
Comment (olensmar) [inline soapui-local-client/src/main/java/com/smartbear/readyapi/client/execution/SoapUIRecipeExecutor.java]: ok - any specific reason?

File: ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/execution/TestServerRecipeExecutionResult.java (hunk 2 above)
Comment (olensmar) [inline ready-api-testserver-client-core/src/main/java/com/smartbear/readyapi/client/execution/TestServerRecipeExecutionResult.java]: agreed

**Comment by prakashmrec** [comment]
@olensmar, @spindelmanne - is this PR ready for merge or waiting for more changes?

**Comment by spindelmanne** [comment]
I'll merge this now but will make a PR of my own with changes I suggeste.

**Pull request merged**